.github/workflows/release.yml

```diff
@@ -0,0 +1,46 @@
+name: Tag & Release
+
+on:
+  push:
+    branches:
+      - 'main'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+
+      - name: Install dev dependencies
+        run: python -m pip install --upgrade pip setuptools poetry
+
+      - name: Build binary wheel and a source tarball
+        run: poetry build
+
+      - name: Get version
+        run: echo "VERSION=$(poetry version --short)" >>$GITHUB_OUTPUT
+        id: version
+
+      - name: Tag with updated version
+        id: tag_version
+        uses: mathieudutour/github-tag-action@v6.1
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          custom_tag: ${{ steps.version.outputs.VERSION }}
```

Comment:
what secrets are these?

Reply:
it's the default github token assigned to every action, it allows the action to push a release